ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ExecuteQueryResponseHandler.java

```diff
@@ -58,31 +62,50 @@
   }
 
   @Override
-  protected void handleMetadata(final QueryResponseMetadata queryResponseMetadata) {
-    cf.queryID().complete(queryResponseMetadata.queryId);
-    columnNames = queryResponseMetadata.columnNames;
-    columnTypes = RowUtil.columnTypesFromStrings(queryResponseMetadata.columnTypes);
+  protected void handleMetadata(final StreamedRow queryResponseMetadata) {
+    final String queryId = queryResponseMetadata.getHeader().get().getQueryId().toString();
+
+    cf.queryID().complete(queryId);
+
+    final List<Column> columns = queryResponseMetadata
+            .getHeader()
+            .get()
+            .getSchema()
+            .columns();
+
+    columnNames = columns
+            .stream()
+            .map(col -> col.name().text())
+            .collect(Collectors.toList());
+
+    final List<String> columnTypesString = columns
+            .stream()
+            .map(col -> col.type().toString())
+            .collect(Collectors.toList());
+
+    columnTypes = RowUtil.columnTypesFromStrings(columnTypesString);
+
     columnNameToIndex = RowUtil.valueToIndexMap(columnNames);
   }
 
   @Override
-  protected void handleRow(final Buffer buff) {
-    final Row row;
-    final Object json = buff.toJson();
-
-    if (json instanceof JsonObject) {
-      final JsonObject jsonObject = (JsonObject) json;
+  protected void handleRow(final StreamedRow row) {
+    if (row.getConsistencyToken().isPresent()) {
       // This is the serialized consistency vector
       // Don't add it to the result list since the user should not see it
-      if (jsonObject.getMap() != null && jsonObject.getMap().containsKey("consistencyToken")) {
-        log.info("Response contains consistency vector " + jsonObject);
-        serializedConsistencyVector.set((String) ((JsonObject) json).getMap().get(
-            "consistencyToken"));
-      } else {
-        throw new RuntimeException("Could not decode JSON, expected consistency toke: " + json);
+      final ConsistencyToken consistencyToken = row.getConsistencyToken().get();
+      try {
+        final String consistencyTokenString = consistencyToken.getConsistencyToken();
+        log.info("Response contains consistency vector " + consistencyTokenString);
+        serializedConsistencyVector.set(consistencyTokenString);
+      } catch (Exception e) {
```

Comment:
Do you need this catch here?  There is no parsing in particular that's happening and no known exception to catch.  I think the error before was that if there was a JsonObject, it must be a consistency token, but using `StreamedRow` takes care of much of the decisioning for you here.

Reply:
yes, you are right. removed `try...catch` block